smoketest.py

```diff
@@ -26,11 +26,11 @@
 
 
 def validate_args():
-    # Ensure that URL is valid and PAT is well formed
-    # TO DO
+    if ((target[0:8] == "https://") and (target[-4] == ".") and (target[-3:].isalnum())):
+        if ((len(pat) == 40) and (pat[0:4] == 'ghp_') and (pat[5:40].isalnum())):
```

Comment:
I'm wondering if we should just wrap this in `get_pat_user()`. We're going to have variable length/format PATs as GHES hasn't implemented the new format (`ghp_`) in all versions. 

We're enforcing the existence of a PAT as an argument, but I'm wondering if the best _validation_ would be seeing if we can actually auth with this...

Reply:
I added a check along these lines ^ at https://github.com/maclarel/ghes-smoketest/blob/master/smoketest.py#L54. Definitely open to recommendations for improvements on this.

Reply:
ah that works then, I didn't think about previous versions. Typically I prefer to do validation locally rather than throwing it at a URL but since it's variable that's definitely the right path!